scripts/devops_tasks/update_sample_code.py

```diff
@@ -0,0 +1,54 @@
+import os
+import re
+import argparse
+
+from ci_tools.functions import discover_targeted_packages
+from ci_tools.parsing import ParsedSetup
+
+root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
+
+def replace_samples_in(directory: str) -> None:
+    # Define the strings to search and replace
+    old_string = "DefaultAzureCredential()"
+    new_string = "DefaultAzureCredential(exclude_managed_identity_credential=True)"
```

Comment:
We don't want the sample to contain this, as it is not the right method for most of our customers.

Reply:
Just wanted to confirm that we only do the replacement in samples folder, right?

Reply:
Yar! We discover all the valid packages under the target sdk folder, then only check `<package-root>/samples` and `<package-root>/sample`

Reply:
AND, for even more confidence, none of the changes made in `Test` can be propogated, as we ship the packages built during `Build` only.